core/src/genesis.rs

```diff
@@ -387,4 +486,26 @@ mod tests {
         )?;
         Ok(())
     }
+
+    #[allow(clippy::unwrap_used)]
+    #[test]
+    fn genesis_block_builder_example() {
+        let public_key = "ed0120204e9593c3ffaf4464a6189233811c297dd4ce73aba167867e4fbd4f8c450acb";
+        let mut genesis_builder = RawGenesisBlockBuilder::new();
+
+        genesis_builder = genesis_builder
+            .domain("wonderland".parse().unwrap())
+            .with_account("alice".parse().unwrap())
+            .with_account("bob".parse().unwrap())
+            .finish_domain()
+            .domain("tulgey_wood".parse().unwrap())
+            .with_account("Cheshire_Cat".parse().unwrap())
+            .finish_domain()
+            .domain("meadow".parse().unwrap())
+            .with_account_and_public_key("Mad_Hatter".parse().unwrap(), public_key.parse().unwrap())
+            .with_asset("hats".parse().unwrap(), AssetValueType::BigQuantity, true)
+            .finish_domain();
```

Comment:
It could, however I think the errors will be clearer if kept outside. Because if it's a parse error you want to fail when you hit
that line not find out that something went wrong at a later time.